### ISSUE TITLE
Errors in multiple MSBuild properties initialization of System.Private.CoreLib.csproj in VS2017

### ISSUE THREAD

**Issue by 4creators** [dotnet/coreclr#12355]
### Building from Visual Studio 2017 command line
According to my tests it affects only builds performed from Visual Studio 2017 command line. Builds done from Visual Studio 2015 command line are not affected.

System.Private.CoreLib.csproj build is invoked via $(RepoRoot)/src/build.proj together with SOS project. During build at the initial stage of property evaluation in System.Private.CoreLib.csproj when majority of properties were not evaluated project [imports Microsoft.Net.Compilers.props at line 4](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/System.Private.CoreLib.csproj#L4):
```
<Import Project="..\..\Tools\net46\roslyn\build\Microsoft.Net.Compilers.props" Condition="'$(OS)'=='Windows_NT'" />
```
Next on [line 7 project imports $(RepoRoot)/dir.props](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/System.Private.CoreLib.csproj#L7):
```
<Import Project="..\..\dir.props" /> 
```
which itself imports [Microsoft.Net.Compilers.props at line 186](https://github.com/dotnet/coreclr/blob/master/dir.props#L186)
```
 <Import Project="$(RoslynPropsFile)" Condition="'$(RunningOnUnix)'!='true' and Exists('$(RoslynPropsFile)') and '$(UseRoslynCompilers)'!='false'" />
```
Resulting double import of Microsoft.Net.Compilers.props file leads to MSBuild warning MSB4011:
```
e:\src\ms\coreclrbug\dir.props(186,3): warning MSB4011: "e:\src\ms\coreclrbug\Tools\net46\roslyn\build\Microsoft.Net.Compilers.props" cannot be imported again. It was already imported at "e:\src\ms\coreclrbug\src\mscorlib\System.Private.CoreLib.csproj (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. 
```
The most important part of this message `This subsequent import will be ignored` indicates that evaluation of MSBuild properties in Microsoft.Net.Compilers.props file will be performed only once before all definitions and property values declared in $(RepoRoot)/dir.props will be introduced into evaluation scope. As a result any property which was evaluated in Microsoft.Net.Compilers.props before $(RepoRoot)/dir.props were imported and was not explicitly overriden will retain it's value. Some of this properties include IntermediateOutputPath.

This set of values results in several side effects for System.Private.CoreLib build which include:

IntermediateOutputPath is set to obj/$(BuildOS).$(Platform).$(Configuration)/System.Private.CoreLib/ relative to System.Private.CoreLib.csproj file i.e. command `build all x86 Release` will give intermediate path as above, although, final build products are copied to expected target directory i.e. `$(RepoRoot)/bin/Product/Windows_NT.x86.Release`

IntermediateOutputPath expressed relative to repo root in this case is as follows:
```
$(RepoRoot)/src/mscorlib/obj/Windows_NT.x86.Release/System.Private.CoreLib/
```
As far as I have checked compiler is invoked correctly with configuration, platform and debug output set as expected for given project build configuration.

### Building from Visual Studio 2017 IDE using System.Private.CoreLib.sln

In this case `System.Private.CoreLib.csproj` is cut off from properties set via $(RepoRoot)/build.proj and $(RepoRoot)/src/build.proj which are used to invoke build from command line. Therefore even more properties are set to their default values including `Platform=x64` and `Configuration=Debug` for choice of directories. This results in that all intermediate and final build artifacts land respectively in:
```
$(RepoRoot)/src/mscorlib/obj/Windows_NT.x64.Debug/System.Private.CoreLib/
```
```
$(RepoRoot)/bin/Product/Windows_NT.x64.Debug/System.Private.CoreLib/
```
No other intermediate and target directories are used.

If one attempts build for Configuration=Checked, Platform=x86 resulting System.Private.CoreLib.dll built with compiler settings as expected for Checked, x86 build will land in the above directories. If one changes configuration to i.e. Release, x64 and runs Build command the result is as below:
```
========== Build: 0 succeeded, 0 failed, 1 up-to-date, 0 skipped ==========
```
In next step running Rebuild we build Release, x64 and build artifacts again land in the Windows_NT.x64.Debug directories. 

### Artifacts
Files with examples of the diagnostic MSBuild output for mistargeted builds in VS 2017 are attached.
[System.Private.CoreLib.csproj.builds.zip](https://github.com/dotnet/coreclr/files/1083395/System.Private.CoreLib.csproj.builds.zip)

**Comment by 4creators** [comment]
Bug was introduced with PR #10430 which was a fix of #10056

**Comment by 4creators** [comment]
Fixed with PR #12356
